File: 515-find-largest-value-in-each-tree-row/515-find-largest-value-in-each-tree-row.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        return solve(root);
    }
    
private:
    vector<int> solve(TreeNode* root) {
        if(root == nullptr)
            return {};
        queue<TreeNode*> queue;
        vector<int> ans;
        
        queue.push(root);
        
        while(!queue.empty()) {
            int levelSize = queue.size();
            int maxElem = INT_MIN;
            
            while(levelSize--) {
                auto currNode = queue.front();
                queue.pop();
                
                maxElem = max(maxElem, currNode -> val);
                
                if(currNode -> left != nullptr)
                    queue.push(currNode -> left);
                if(currNode -> right != nullptr)
                    queue.push(currNode -> right);
            }
            ans.push_back(maxElem);
        }
        return ans;
    }
};

File: 515-find-largest-value-in-each-tree-row/515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        if root == None:
            return []
        queue = deque()
        ans = []
        
        queue.append(root)
        
        while queue:
            levelSize = len(queue)
            maxElem = -float('inf')
            
            for _ in range(levelSize):
                currNode = queue.popleft()
                
                maxElem = max(maxElem, currNode.val)
                
                if currNode.left != None:
                    queue.append(currNode.left)
                if currNode.right != None:
                    queue.append(currNode.right)
            ans.append(maxElem)
        return ans
        